src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/TextFieldParser.vb

```diff
@@ -0,0 +1,1576 @@
+
+        ''' <summary>
+        '''  The field being built
+        ''' </summary>
+        ''' <value>The field</value>
+        Public ReadOnly Property Field() As String
+            Get
+                Return m_Field.ToString()
+            End Get
+        End Property
+
+        ''' <summary>
+        '''  The current index on the line. Used to indicate how much of the line was used to build the field
+        ''' </summary>
+        ''' <value>The current position on the line</value>
+        Public ReadOnly Property Index() As Integer
+            Get
+                Return m_Index
+            End Get
+        End Property
+
+        ''' <summary>
+        '''  The length of the closing delimiter if one was found
+        ''' </summary>
+        ''' <value>The length of the delimiter</value>
+        Public ReadOnly Property DelimiterLength() As Integer
+            Get
+                Return m_DelimiterLength
+            End Get
+        End Property
+
+
+        ''' <summary>
+        '''  Indicates that the current field breaks the subset of csv rules we enforce
+        ''' </summary>
+        ''' <value>True if the line is malformed, otherwise False</value>
+        ''' <remarks>
+        '''  The rules we enforce are:
+        '''      Embedded quotes must be escaped
+        '''      Only space characters can occur between a delimiter and a quote
+        '''</remarks>
+        Public ReadOnly Property MalformedLine() As Boolean
+            Get
+                Return m_MalformedLine
+            End Get
+        End Property
+
+        ''' <summary>
+        '''  Builds a field by walking through the passed in line starting at StartAt
+        ''' </summary>
+        ''' <param name="Line">The line containing the data</param>
+        ''' <param name="StartAt">The index at which we start building the field</param>
+        Public Sub BuildField(ByVal Line As String, ByVal StartAt As Integer)
+            m_Index = StartAt
+            Dim Length As Integer = Line.Length
+
+            While m_Index < Length
+
+                If Line(m_Index) = DoubleQuoteChar Then
+
+                    ' Are we at the end of the file?
+                    If m_Index + 1 = Length Then
+                        ' We've found the end of the field
+                        m_FieldFinished = True
+                        m_DelimiterLength = 1
+
+                        ' Move index past end of line
+                        m_Index += 1
+                        Return
+                    End If
+                    ' Check to see if this is an escaped quote
+                    If m_Index + 1 < Line.Length And Line(m_Index + 1) = DoubleQuoteChar Then
+                        m_Field.Append(DoubleQuoteChar)
+                        m_Index += 2
+                        Continue While
+                    End If
+
+                    ' Find the next delimiter and make sure everything between the quote and
+                    ' the d
```

Comment:
Consider moving the constant into the one method above that uses it. Also, typo in `readability`.